plugins/com.google.cloud.tools.eclipse.preferences/src/com/google/cloud/tools/eclipse/preferences/Activator.java

```diff
@@ -41,4 +46,20 @@ public static Activator getDefault() {
   public static ImageDescriptor getImageDescriptor(String path) {
     return imageDescriptorFromPlugin(PLUGIN_ID, path);
   }
+
+  /**
+   * This methods is not supposed to be overridden, but we couldn't find any alternative
+   * to replace the Instance-scoped {@link AbstractUIPlugin#preferenceStore} in the base
+   * class with a Configuration-scoped {@link ScopedPreferenceStore}. As such, this class
+   * defines its own {@link preferenceStore} field that is distinct from the field by the
+   * same name in the base class.
+   */
+  @Override
+  public IPreferenceStore getPreferenceStore() {
+    if (preferenceStore == null) {
```

Comment:
Right, we are mostly likely good with one final field of ours. I just copy-pasted the code from the base class. I will take off the lazy initialization.